internal/k8s/reconciler/validator/gateway.go

```diff
@@ -380,7 +380,7 @@ func (g *GatewayValidator) validateTLS(ctx context.Context, state *state.Listene
 	} else if !allowed {
 		nsName := getNamespacedName(ref.Name, ref.Namespace, gateway.Namespace)
 		//g.logger.Warn("Cross-namespace listener certificate not allowed without matching ReferenceGrant", "refName", nsName.Name, "refNamespace", nsName.Namespace)
-		state.Status.ResolvedRefs.InvalidCertificateRef = rerrors.NewCertificateResolutionErrorNotPermitted(
+		state.Status.ResolvedRefs.RefNotPermitted = rerrors.NewCertificateResolutionErrorNotPermitted(
```

Comment:
Are we also supposed to return `RefNotPermitted` if the secret referenced does not exist (regardless of ReferenceGrant existence)? Curious if that case is still supposed to use the old error code and, if so, are we respecting that here?

Reply:
Ignore the above, you can see us still using `InvalidCertificateRef` in the code below this change when we can't find the Secret